src/Cli/dotnet/commands/RestoringCommand.cs

```diff
@@ -76,19 +76,97 @@ private static IEnumerable<string> Prepend(string argument, IEnumerable<string>
+        // these properties trigger a separate restore
+        private static readonly string[] PropertiesToExcludeFromRestore =
+        [
+            "TargetFramework"
+        ];
+
+        private static readonly string[] FlagsThatTriggerSilentRestore =
+            [
+                "getProperty",
+                "getItem",
+                "getTargetResult"
+            ];
 
         //  These arguments don't by themselves require that restore be run in a separate process,
         //  but if there is a separate restore process they shouldn't be passed to it
+        private static readonly string[] FlagsToExcludeFromRestore =
+        [
+            ..FlagsThatTriggerSilentRestore,
+            "t",
+            "target",
+            "consoleloggerparameters",
+            "clp"
+        ];
+
+        private static List<string> FlagsToExcludeFromSeparateRestore =
+            ComputeFlags(FlagsToExcludeFromRestore).ToList();
+
+        private static List<string> FlagsThatTriggerSilentSeparateRestore =
+            ComputeFlags(FlagsThatTriggerSilentRestore).ToList();
+
+        private static List<string> PropertiesToExcludeFromSeparateRestore =
+            ComputePropertySwitches(PropertiesToExcludeFromRestore).ToList();
+
+        // We investigate the arguments we're about to send to a separate restore call and filter out
+        // arguments that negatively influence the restore. In addition, some flags signal different modes of execution
+        // that we need to compensate for, so we might yield new arguments that should be included in the overall restore call.
+        private static (string[] newArgumentsToAdd, string[] existingArgumentsToForward) ProcessForwardedArgumentsForSeparateRestore(IEnumerable<string> forwardedArguments)
+        {
+            HashSet<string> newArgumentsToAdd = new();
+            List<string> existingArgumentsToForward = new();
+
+            foreach (var argument in forwardedArguments)
+            {
+
+                if (!IsExcludedFromSeparateRestore(argument) && !IsExcludedFromRestore(argument))
+                {
+                    existingArgumentsToForward.Add(argument);
+                }
+
+                if (TriggersSilentSeparateRestore(argument))
+                {
+                    newArgumentsToAdd.Add("-nologo");
+                    newArgumentsToAdd.Add("-verbosity:quiet");
+                }
+            }
+            return (newArgumentsToAdd.ToArray(), existingArgumentsToForward.ToArray());
+        }
+        private static IEnumerable<string> ComputePropertySwitches(string[] properties)
+        {
+            foreach (var prefix in switchPrefixes)
+            {
+                foreach (var property in properties)
+                {
+                    yield return $"{prefix}property:{property}=";
+                    yield return $"{prefix}p:{property}=";
+                }
+            }
```

Comment:
Just for fun, I was trying to remember how to write this as LINQ. Just did a small test app, and it would be written like this:
```C#
return switchPrefixes.SelectMany(prefix => properties.SelectMany(prop => new string[] { $"{prefix}property:{prop}=", $"{prefix}p:{prop}=" }))
```

I don't think you can use the `[]` syntax for the string array since it isn't aware of the type in the lambda.